src/Static.jl

```diff
@@ -430,6 +430,9 @@ Base.any(::Tuple{Vararg{True}}) = true
 Base.any(::Tuple{Vararg{Union{True, False}}}) = true
 Base.any(::Tuple{Vararg{False}}) = false
 
+Base.real(@nospecialize(x::StaticInteger)) = x
+Base.imag(@nospecialize(_::StaticInteger)) = Zero()
```

Comment:
```suggestion
Base.imag(@nospecialize(x::StaticNumber)) = zero(x)
```
This allows it to work for all static numbers.